### PR TITLE
931 Update Nox.Generator to allow saving files with System.IO.File

### LINKED ISSUE #931: [Nox.Generator] [Nox.Ui] Update Nox.Generator to allow saving files with System.IO.File
Based on #773 investigation, extend Nox.Generator to allow saving files with System.IO.File instead of using current context flow.

FYI @rochar @DanielHodgeGH

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

File: src/Nox.Generator/Common/TemplateCodeBuilder.cs [hunk 1]
```diff
@@ -136,10 +150,23 @@ private void GenerateSourceCode(string template, object model, string sourceFile
...
-        _context.AddSource(sourceFileName,
-            SourceText.From(sourceText,
-            Encoding.UTF8));
+
+        if (string.IsNullOrEmpty(_fileExtension) || _fileExtension == "cs")
+        {
+            _context.AddSource($"{sourceFileName}.g.cs", SourceText.From(sourceText, Encoding.UTF8));
+        }
+        else
+        {
```
Comment (anna-naumova25) [inline src/Nox.Generator/Common/TemplateCodeBuilder.cs]: This is a very rough implementation. We should probably configure output path on the client side, maybe in appsetings.json

File: src/Nox.Generator/Common/TemplateCodeBuilder.cs
Comment (rochar) [inline src/Nox.Generator/Common/TemplateCodeBuilder.cs:1]: @anna-naumova25 having a new generator/template we should have the new generated files, in this case in Crypto Cash UI, to be easy to validate the teplate

File: src/Nox.Generator/Common/TemplateCodeBuilder.cs (hunk 1 above)
Comment (rochar) [reply src/Nox.Generator/Common/TemplateCodeBuilder.cs]: @anna-naumova25  lets think about composition or inheritance to solve this, so we keep 2 generators the simple and with only the specific code.

File: src/Nox.Generator/Common/TemplateCodeBuilder.cs [hunk 2]
```diff
@@ -136,10 +150,23 @@ private void GenerateSourceCode(string template, object model, string sourceFile
...
+#pragma warning disable RS1035 // Do not use APIs banned for analyzers
+            string outputFolder = "Generated.NonCs";
```
Comment (rochar) [inline src/Nox.Generator/Common/TemplateCodeBuilder.cs]:
```suggestion
            string outputFolder = "Nox.Generated";
```

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Generator/Common/TemplateCodeBuilder.cs
Comment (anna-naumova25) [reply src/Nox.Generator/Common/TemplateCodeBuilder.cs:1]: Sure, I've created a separate ticket for it - #932 as there will be lots of changes. So I wanted to keep this PR simple and clear.

File: src/Nox.Generator/Common/TemplateCodeBuilder.cs (hunk 1 above)
Comment (anna-naumova25) [reply src/Nox.Generator/Common/TemplateCodeBuilder.cs]: I've create TemplateCodeBuilderBase and two derived classes - TemplateCodeBuilder and TemplateCodeBuilderFile to separate the logic

File: src/Nox.Generator/Common/TemplateCodeBuilderFile.cs
Comment (rochar) [inline src/Nox.Generator/Common/TemplateCodeBuilderFile.cs:1]: looka much better now

**Review by rochar — APPROVED** [review]

**Review by ILoveMyTortoise — APPROVED** [review]

**Pull request merged**